packages/foundry/contracts/witch/WitchV2.sol

```diff
@@ -456,67 +565,28 @@ contract WitchV2 is AccessControl {
         }
 
         uint256 inkAtEnd = uint256(artIn).wdiv(auction_.art).wmul(auction_.ink);
-        inkOut = inkAtEnd.wmul(proportionNow);
-    }
-
-    /// @notice Update accounting on the Witch and on the Cauldron. Delete the auction and give back the vault if finished.
-    /// This function doesn't verify the vaultId matches the vault and auction passed. Check before calling.
-    function _updateAccounting(
-        bytes12 vaultId,
-        bytes6 ilkId,
-        bytes6 baseId,
-        WitchDataTypes.Auction memory auction_,
-        uint256 inkOut,
-        uint256 artIn
-    ) internal {
-        // Duplicate check, but guarantees data integrity
-        require(auction_.start > 0, "Vault not under auction");
-
-        // Update concurrent collateral under auction
-        WitchDataTypes.Limits memory limits_ = limits[ilkId][baseId];
-
-        // Update local auction
-        {
-            if (auction_.art == artIn) {
-                // If there is no debt left, return the vault with the collateral to the owner
-                _auctionEnded(vaultId, auction_.owner);
-
-                // Update limits - reduce it by the whole auction
-                limits_.sum -= auction_.ink;
-            } else {
-                // Ensure enough dust is left
-                DataTypes.Debt memory debt = cauldron.debt(baseId, ilkId);
-                require(
-                    auction_.art - artIn >= debt.min * (10**debt.dec),
-                    "Leaves dust"
-                );
-
-                // Update the auction
-                auction_.ink -= inkOut.u128();
-                auction_.art -= artIn.u128();
-
-                // Store auction changes
-                auctions[vaultId] = auction_;
-
-                // Update limits - reduce it by whatever was bought
-                limits_.sum -= inkOut.u128();
-            }
+        liquidatorCut = inkAtEnd.wmul(proportionNow);
+        if (auction_.auctioneer != to) {
+            auctioneerCut = liquidatorCut.wmul(auctioneerReward);
+            liquidatorCut -= auctioneerCut;
         }
-
-        // Store limit changes
-        limits[ilkId][baseId] = limits_;
-
-        // Update accounting at Cauldron
-        cauldron.slurp(vaultId, inkOut.u128(), artIn.u128());
     }
 
+    // ======================================================================
+    // =                        Lifecycle functions                         =
+    // ======================================================================
```

Comment:
I would move `_auctionStarted` just after `auction`, `_auctionEnded` just after `cancel` (even if it also gets called in the `pay*` functions) and `_collateralBought` just after `_updateAccounting`. With that this block gets dissolved.